torchaudio/csrc/transducer.cpp

```diff
@@ -0,0 +1,82 @@
+#ifdef BUILD_TRANSDUCER
+
+#include <iostream>
+#include <numeric>
+#include <string>
+#include <vector>
+
+#include <torch/script.h>
+#include "rnnt.h"
+
+int64_t cpu_rnnt_loss(torch::Tensor acts,
+                      torch::Tensor labels,
+                      torch::Tensor input_lengths,
+                      torch::Tensor label_lengths,
+                      torch::Tensor costs,
+                      torch::Tensor grads,
+                      int64_t blank_label,
+                      int64_t num_threads) {
+
+    int maxT = acts.size(1);
+    int maxU = acts.size(2);
+    int minibatch_size = acts.size(0);
+    int alphabet_size = acts.size(3);
+
+    rnntOptions options;
+    memset(&options, 0, sizeof(options));
+    options.maxT = maxT;
+    options.maxU = maxU;
+    options.blank_label = blank_label;
+    options.batch_first = true;
+    options.loc = RNNT_CPU;
+    options.num_threads = num_threads;
+
+    // have to use at least one
+    options.num_threads = std::max(options.num_threads, (unsigned int) 1);
+
+    size_t cpu_size_bytes = 0;
+    switch (acts.scalar_type()) {
+      case torch::ScalarType::Float:
+        {
+        get_workspace_size(maxT, maxU, minibatch_size,
+                           false, &cpu_size_bytes);
+
+        std::vector<float> cpu_workspace(cpu_size_bytes / sizeof(float), 0);
+
+        compute_rnnt_loss(acts.data<float>(), grads.data<float>(),
+                          labels.data<int>(), label_lengths.data<int>(),
+                          input_lengths.data<int>(), alphabet_size,
+                          minibatch_size, costs.data<float>(),
+                          cpu_workspace.data(), options);
```

Comment:
Note `data()` is issuing a bunch of deprecation warnings;

```
/scratch/moto/torchaudio/torchaudio/csrc/transducer.cpp:44:44: warning: ‘T* at::Tensor::data() const [with T = float]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
         compute_rnnt_loss(acts.data<float>(), grads.data<float>(),
                                            ^
In file included from /home/moto/conda/envs/PY3.8-cuda101/lib/python3.8/site-packages/torch/include/ATen/Tensor.h:3:0,
                 from /home/moto/conda/envs/PY3.8-cuda101/lib/python3.8/site-packages/torch/include/ATen/Context.h:4,
                 from /home/moto/conda/envs/PY3.8-cuda101/lib/python3.8/site-packages/torch/include/ATen/ATen.h:9,
                 from /home/moto/conda/envs/PY3.8-cuda101/lib/python3.8/site-packages/torch/include/torch/csrc/api/include/torch/types.h:3,
                 from /home/moto/conda/envs/PY3.8-cuda101/lib/python3.8/site-packages/torch/include/torch/script.h:3,
                 from /scratch/moto/torchaudio/torchaudio/csrc/transducer.cpp:6:
```

Reply:
yes, that's one of the quick follow-up: i opened #1160 to remove this just now. feel free to stamp :)